src/lib/consensus/proof_of_stake.ml

```diff
@@ -2062,12 +2093,14 @@ module Data = struct
         ~(constraint_constants : Genesis_constants.Constraint_constants.t) =
       let max_sub_window_density = constants.slots_per_sub_window in
       let max_window_density = constants.slots_per_window in
-      let blockchain_length =
+      let blockchain_length, global_slot_since_genesis =
         match constraint_constants.fork with
         | None ->
-            Length.zero
-        | Some {previous_length; _} ->
-            previous_length
+            (Length.zero, Coda_numbers.Global_slot.zero)
+        | Some {previous_length; previous_global_slot; _} ->
+            (*Note: global_slot_since_genesis at fork point is the same as global_slot_since_genesis in the new genesis. This value is used to check transaction validity and existence of locked tokens.
+            For reviewers, should this be incremented by 1 because it's technically a new block? we don't really know how many slots passed since the fork point*)
```

Comment:
> should this be incremented by 1 because it's technically a new block?

I think this is okay to leave as is, since IIRC we don't consume `negative_one` anywhere except in the proof and the previous protocol state hash. Both of these are opaque, so the true block and the 'fake' negative one block can never realistically be confused with eachother.

I also find it appealing and easy to reason about if the -1th block of the fork is the last block from before the fork (or, at least, occupies the same point in time).

Reply:
The slot remains the same even for the genesis block and will show up as a new block, say, in archive db. So there will be two different blocks in the main chain with the same` global_slot_since_genesis` (maybe not very different from having multiple genesis blocks with slot = 0 in the main chain as a result of multiple hardforks?)